core/sparse_array.cpp

```diff
@@ -68,6 +87,31 @@ SparseArray::~SparseArray(void)
 	
 	free(strengths_);
 	strengths_ = nullptr;
+
+    //Deallocating all multithreading parts 
+    for(int i=0;i<nrows_;i++)
+    {
+
+        free(columns[i]);
+        columns[i] = nullptr;
+
+        free(distances[i]);
+        distances[i] = nullptr;
```

Comment:
I do generally like to zero out instance variables in dealloc, so that if somebody downstream tries to use the variables they crash.  Here, though, I wouldn't bother zeroing out columns[i] and distances[i], because we're about to free and zero columns and distances themselves, so these values will all be inaccessible momentarily anyway.  Save yourself nrows_*2 writes to memory, for what that's worth.